system/lib/update_llvm_libc.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright 2019 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import os
+import sys
+import shutil
+import glob
+import subprocess
+
+script_dir = os.path.abspath(os.path.dirname(__file__))
+emscripten_root = os.path.dirname(os.path.dirname(script_dir))
+default_llvm_dir = os.path.join(os.path.dirname(emscripten_root), 'llvm-project')
+emscripten_patches = os.path.join(script_dir, "llvm-libc", "patches", "*.patch")
```

Comment:
We don't tend to check in patches like this, at least not for the other libraries here.  Instead we push our changes out to https://github.com/emscripten-core/llvm-project and maintain the patches in git.

I'm not totally opposed to this approach here, but it would be new thing for us.